File: Python/Task24/n38958.py
# https://inf-ege.sdamgia.ru/problem?id=38958

file = open("n38958.txt", "r")
curLen = lenFromLastA = maxLen = 0
while x := file.read(1): 
	if x == 'A':
		curLen = lenFromLastA + 1
		lenFromLastA = 0
	else:
		curLen += 1
		lenFromLastA += 1
	maxLen = max(curLen, maxLen)
print(maxLen)

File: C++/Task24/n38958/n38958.cpp
// https://inf-ege.sdamgia.ru/problem?id=38958

#include <iostream>
#include <fstream>

int main()
{
	std::fstream myFile(".\\Data\\n38958.txt", std::ios_base::in);
	int curLen = 0, lenFromLastA = 0, maxLen = 0;
	char x;
	while (myFile >> x)
	{
		if (x == 'A')
		{
			curLen = lenFromLastA + 1;
			lenFromLastA = 0;
		}
		else
		{
			curLen++;
			lenFromLastA++;
		}
		maxLen = std::max(curLen, maxLen);
	}
	std::cout << maxLen << std::endl;
	return 0;
}